packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-outbound-section.tsx

```diff
@@ -0,0 +1,469 @@
+    fields: "*prices,+service_zone.fulfillment_set.location.id",
+    /**
+     * TODO: this should accept filter for location_id
+     */
+  })
+
+  const { mutateAsync: addInboundShipping } = useAddClaimInboundShipping(
+    claim.id,
+    order.id
+  )
+
+  const { mutateAsync: deleteInboundShipping } = useDeleteClaimInboundShipping(
+    claim.id,
+    order.id
+  )
+
+  const { mutateAsync: addOutboundItem } = useAddClaimOutboundItems(
+    claim.id,
+    order.id
+  )
+
+  const { mutateAsync: updateOutboundItem } = useUpdateClaimOutboundItems(
+    claim.id,
+    order.id
+  )
+
+  const { mutateAsync: removeOutboundItem } = useRemoveClaimOutboundItem(
+    claim.id,
+    order.id
+  )
+
+  /**
+   * Only consider items that belong to this claim.
+   */
+  const previewOutboundItems = useMemo(
+    () =>
+      preview?.items?.filter(
+        (i) =>
+          !!i.actions?.find(
+            (a) => a.claim_id === claim.id && a.action === "ITEM_ADD"
+          )
+      ),
+    [preview.items]
+  )
+
+  const variantItemMap = useMemo(
+    () => new Map(order?.items?.map((i) => [i.variant_id, i])),
+    [order.items]
+  )
+
+  const {
+    fields: outboundItems,
+    append,
+    remove,
+    update,
+  } = useFieldArray({
+    name: "outbound_items",
+    control: form.control,
+  })
+
+  const variantOutboundMap = useMemo(
+    () => new Map(previewOutboundItems.map((i) => [i.variant_id, i])),
+    [previewOutboundItems, outboundItems]
+  )
+
+  useEffect(() => {
+    const existingItemsMap: Record<string, boolean> = {}
+
+    previewOutboundItems.forEach((i) => {
+      const ind = outboundItems.findIndex((field) => field.item_id === i.id)
+
+      existingItemsMap[i.id] = true
+
+      if (ind > -1) {
+        if (outboundItems[ind].quantity !== i.detail.quantity) {
+          update(ind, {
+            ...outboundItems[ind],
+            quantity: i.detail.quantity,
+          })
+        }
+      } else {
+        append({
+          item_id: i.id,
+          quantity: i.detail.quantity,
+          variant_id: i.variant_id,
+        })
+      }
+    })
+
+    outboundItems.forEach((i, ind) => {
+      if (!(i.item_id in existingItemsMap)) {
+        remove(ind)
+      }
+    })
+  }, [previewOutboundItems])
+
+  useEffect(() => {
+    // TODO: Pick the shipping methods from actions where return_id is null for outbound
+    const method = preview.shipping_methods.find(
+      (s) => !!s.actions?.find((a) => a.action === "SHIPPING_ADD")
+    )
+
+    if (method) {
+      form.setValue("outbound_option_id", method.shipping_option_id)
+    }
+  }, [preview.shipping_methods])
+
+  const locationId = form.watch("location_id")
+  const showOutboundItemsPlaceholder = !outboundItems.length
+
+  const onItemsSelected = async () => {
+    itemsToAdd.length &&
+      (await addOutboundItem(
+        {
+          items: itemsToAdd.map((variantId) => ({
+            // TODO: Line item will not be present at this stage. Do we skip this?
```

Comment:
Yeah, I don't think we'll have a line item here – maybe @carlos-r-l-rodrigues can confirm?

Reply:
It's not clear to me which stage this is, but the line item will be create once the action the endpoint `POST /admin/claims/[id]/outbound/items` is called, and it will be returned on the order preview

Reply:
Yeah, this is the action doing exactly that. So the payload should not/can't contain line item data.

Reply:
Sounds good, removed this from there